### PR TITLE
*: supports delete store label

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve?

Issue Number: Close https://github.com/tikv/pd/issues/5509

### What is changed and how does it work?

```commit-message
*: supports store command to delete the specified label
```

### Check List

Tests

- Unit test

### Release note

```release-note
store command of pdctl supports to delete the specified label for a store
```

### CODE REVIEW

File: server/cluster/cluster.go [hunk 1]
```diff
@@ -1046,17 +1046,47 @@ func (c *RaftCluster) GetRangeHoles() [][]string {
...
+	} else {
+		// If 'force' isn't set, the given labels will merge into those labels which already existed in the store.
+		newStore.Labels = core.MergeLabels(newStore.GetLabels(), labels)
+	}
+	// PutStore will perform label merge.
+	return c.putStoreImpl(newStore)
+}
+
+// DeleteStoreLabel updates a store's location labels
+// If 'force' is true, then update the store's labels forcibly.
```
Comment (CabinfeverB) [inline server/cluster/cluster.go:1063]: pls remove it

File: tools/pd-ctl/pdctl/command/store_command.go [hunk 2]
```diff
@@ -499,13 +514,60 @@ func labelStoreCommandFunc(cmd *cobra.Command, args []string) {
 		return
 	}
 	prefix := fmt.Sprintf(path.Join(storePrefix, "label"), args[0])
+	var toDeletedLabel string
+	if isDelete {
+		if len(args) != 2 {
+			cmd.PrintErrln("Failed: not allow to delete multiple labesl at a time")
```
Comment (rleungx) [inline tools/pd-ctl/pdctl/command/store_command.go:520]:
```suggestion
			cmd.PrintErrln("Failed: not allow to delete multiple labels at a time")
```

File: tools/pd-ctl/pdctl/command/store_command.go [hunk 3]
```diff
@@ -490,7 +500,12 @@ func labelStoreCommandFunc(cmd *cobra.Command, args []string) {
 	// The least args' numbers is 1, which means users can set empty key and value
 	// In this way, if force flag is set then it means clear all labels,
 	// if force flag isn't set then it means do nothing
-	if len(args) < 1 || len(args)%2 != 1 {
+	isDelete, err := cmd.Flags().GetBool("delete")
+	if err != nil {
+		cmd.Println("can't parse paramenters", err)
```
Comment (rleungx) [inline tools/pd-ctl/pdctl/command/store_command.go:505]:
```suggestion
		cmd.Println("can't parse parameters", err)
```

File: tools/pd-ctl/pdctl/command/store_command.go [hunk 4]
```diff
@@ -499,13 +514,60 @@ func labelStoreCommandFunc(cmd *cobra.Command, args []string) {
...
+	if isDelete {
+		if len(args) != 2 {
+			cmd.PrintErrln("Failed: not allow to delete multiple labesl at a time")
+			return
+		}
+		toDeletedLabel = args[1]
+		b, _ := json.Marshal(toDeletedLabel)
+		res, err := doRequest(cmd, prefix, http.MethodDelete, http.Header{"Content-Type": {"application/json"}}, WithBody(bytes.NewBuffer(b)))
+		if err != nil {
+			cmd.Printf("Failed! %s", err)
```
Comment (rleungx) [inline tools/pd-ctl/pdctl/command/store_command.go:527]:
```suggestion
			cmd.Printf("Failed! %s\n", err)
```

File: tools/pd-ctl/pdctl/command/store_command.go [hunk 5]
```diff
@@ -499,13 +514,60 @@ func labelStoreCommandFunc(cmd *cobra.Command, args []string) {
...
 	labels := make(map[string]interface{})
-	for i := 1; i < len(args); i += 2 {
-		labels[args[i]] = args[i+1]
+	// useEqual is used to compatible the old way
+	// TODO: remove old way
+	useEqual := true
+	for i := 1; i < len(args); {
+		if i == 1 {
+			sliceArg := strings.Split(args[i], "=")
+			fmt.Println("judge way", sliceArg)
```
Comment (rleungx) [inline tools/pd-ctl/pdctl/command/store_command.go:541]: remove it

File: tools/pd-ctl/pdctl/command/store_command.go [hunk 6]
```diff
@@ -499,13 +514,60 @@ func labelStoreCommandFunc(cmd *cobra.Command, args []string) {
...
+			if len(sliceArg) != 2 {
+				useEqual = false
+			}
+		}
+		if useEqual { // new way
+			sliceArg := strings.Split(args[i], "=")
+			fmt.Println("new way", sliceArg)
```
Comment (rleungx) [inline tools/pd-ctl/pdctl/command/store_command.go:548]: ditto

File: server/cluster/cluster.go [hunk 7]
```diff
@@ -1046,17 +1046,46 @@ func (c *RaftCluster) GetRangeHoles() [][]string {
...
+	labels := make([]*metapb.StoreLabel, 0, len(newStore.GetLabels()))
+	hit := false
+	for _, label := range newStore.GetLabels() {
+		if label.Key == labelKey {
+			hit = true
+			continue
+		}
+		labels = append(labels, label)
+	}
+	if !hit {
```
Comment (disksing) [inline server/cluster/cluster.go:1078]: I think we don't need `hit`, can check `if len(labels) == len(newStore.Labels)` instead.

File: server/cluster/cluster.go [hunk 8]
```diff
@@ -1046,17 +1046,46 @@ func (c *RaftCluster) GetRangeHoles() [][]string {
...
+}
+
+// DeleteStoreLabel updates a store's location labels
+func (c *RaftCluster) DeleteStoreLabel(storeID uint64, labelKey string) error {
+	store := c.GetStore(storeID)
+	if store == nil {
+		return errs.ErrInvalidStoreID.FastGenByArgs(storeID)
+	}
+	newStore := proto.Clone(store.GetMeta()).(*metapb.Store)
+	labels := make([]*metapb.StoreLabel, 0, len(newStore.GetLabels()))
```
Comment (disksing) [inline server/cluster/cluster.go:1069]:
```suggestion
	labels := make([]*metapb.StoreLabel, 0, len(newStore.GetLabels())-1)
```

**Comment by nolouch** [comment]
PTAL @rleungx @CabinfeverB

**Comment by nolouch** [comment]
ptal @disksing

**Review by rleungx — APPROVED** [review]

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**